ide/commons-gwt/src/main/java/org/eclipse/che/ide/rest/UrlBuilder.java

```diff
@@ -37,7 +38,9 @@ public UrlBuilder(String url) {
         JSONObject o = new JSONObject(jso);
         setHost(o.get("host").isString().stringValue());
         setProtocol(o.get("protocol").isString().stringValue());
-        setPort(Integer.valueOf(o.get("port").isString().stringValue()));
+        if(o.containsKey("port") && !Strings.isNullOrEmpty(o.get("port").isString().stringValue())) {
+            setPort(Integer.valueOf(o.get("port").isString().stringValue()));
```

Comment:
In lines 41 and 42 you're using `o.get("port").isString().stringValue()`.
Pay attention that method `JSONObject#get` not as simple as it may seem.
I think it would be nice to move `o.get("port").isString().stringValue()` to the local variable.
Also it would made these two lines much more simpler for reading.
WDYT?